packages/design-system/src/lib/stories/file-uploader/FileUploader.stories.tsx

```diff
@@ -0,0 +1,39 @@
+import type { Meta, StoryObj } from '@storybook/react'
+import { FileUploadState, FileUploader } from '../../components/file-uploader/FileUploader'
+
+/**
+ * ## Description
+ * A file uploader allows users to upload files.
+ */
+const meta: Meta<typeof FileUploader> = {
+  title: 'File uploader',
+  component: FileUploader,
+  tags: ['autodocs']
+}
+
+export default meta
+type Story = StoryObj<typeof FileUploader>
+
+export const Default: Story = {
+  render: () => (
+    <FileUploader
+      upload={function (files: File[]): Map<string, FileUploadState> {
```

Comment:
It went in a slightly different direction, I moved the state one component up. It will be a shared state between two components; there is also an uploaded files list at the bottom with a save button (not implemented yet). Anyway, it is more finished now and should be clearer where I was going with this. I tested it with several thousands of files at once, it handles it well. Should be more than enough for the intended use case. We might want to add something to limit how many files are uploaded at once, if not implemented yet in the jsdv library.